src/dlmalloc.rs

```diff
@@ -4,12 +4,13 @@
 // The original source was written by Doug Lea and released to the public domain
 
 use core::cmp;
+use core::marker::PhantomData;
 use core::mem;
 use core::ptr;
 
-use sys;
+use System;
 
-pub struct Dlmalloc {
+pub struct Dlmalloc<S> {
```

Comment:
Could this perhaps be `S = SomethingDefaulted` to general consumers don't have to worry about this? (platforms like wasm and Linux won't really have another implementation).

On platforms which don't have anything supported `SomethingDefaulted` could be implemented to return errors.

Reply:
I definitely agree with what your trying to achieve; the impact of the changes should be minimal on existing users (with the exception of the SGX platform). But that already is implemented in [https://github.com/alexcrichton/dlmalloc-rs/pull/15/commits/63ba62b32132c8e3ef7e4102e548dedcfdb38968#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R72-R80](lib.rs) This `Dlmalloc` type is private to the crate.